src/main/kotlin/com/cognifide/gradle/aem/common/instance/Instance.kt

```diff
@@ -2,34 +2,45 @@ package com.cognifide.gradle.aem.common.instance
 
 import com.cognifide.gradle.aem.AemException
 import com.cognifide.gradle.aem.AemExtension
+import com.cognifide.gradle.aem.common.instance.action.AwaitDownAction
+import com.cognifide.gradle.aem.common.instance.action.AwaitUpAction
+import com.cognifide.gradle.aem.common.instance.action.CheckAction
+import com.cognifide.gradle.aem.common.instance.action.ReloadAction
 import com.cognifide.gradle.common.utils.Formats
 import com.cognifide.gradle.common.utils.Patterns
 import com.cognifide.gradle.common.utils.formats.JsonPassword
 import com.fasterxml.jackson.annotation.JsonIgnore
 import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import org.apache.commons.lang3.builder.EqualsBuilder
+import org.apache.commons.lang3.builder.HashCodeBuilder
 import java.io.Serializable
 import java.time.ZoneId
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Internal
 
-interface Instance : Serializable {
+open class Instance(@Transient @JsonIgnore protected val aem: AemExtension) : Serializable {
```

Comment:
flattened inheritance

previous: Instance -> AbstractInstance -> [Local|Remote]Instance
now: RemoteInstance -> Instance

why? `instance.satisfy()` was in `AbstractInstance` so that it was not available in provisional step sync callback